### PR TITLE
Enable ballast configuration cluster in all-clusters-app

### PR DESCRIPTION
**Pull request by markus-becker-tridonic-com**
Ballast Configuration cluster was available in the data model XML, but wasn't enabled in any example app. controller-app has the client side already enabled.

### CODE REVIEW

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.matter [hunk 1]
```diff
@@ -3067,6 +3067,19 @@ server cluster ColorControl = 768 {
   command StepColorTemperature(StepColorTemperatureRequest): DefaultSuccess = 76;
 }
 
+server cluster BallastConfiguration = 769 {
+  readonly attribute int8u physicalMinLevel = 0;
+  readonly attribute int8u physicalMaxLevel = 1;
+  attribute int8u minLevel = 16;
+  attribute int8u maxLevel = 17;
+  readonly attribute int8u lampQuantity = 32;
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/all-clusters-common/all-clusters-app.matter:3075]: Looks like this is not enabling any of the optional attributes, right?  Should probably enable those for all-clusters-app, and enable this cluster with only mandatory attributes for all-clusters-minima-app.

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.matter (hunk 1 above)
Comment (markus-becker-tridonic-com) [reply examples/all-clusters-app/all-clusters-common/all-clusters-app.matter]: done

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.matter [hunk 2]
```diff
@@ -3067,6 +3067,28 @@ server cluster ColorControl = 768 {
...
+server cluster BallastConfiguration = 769 {
+  readonly attribute int8u physicalMinLevel = 0;
+  readonly attribute int8u physicalMaxLevel = 1;
+  readonly attribute bitmap8 ballastStatus = 2;
+  attribute int8u minLevel = 16;
+  attribute int8u maxLevel = 17;
+  attribute nullable int8u intrinsicBalanceFactor = 20;
+  attribute nullable int8u ballastFactorAdjustment = 21;
+  readonly attribute int8u lampQuantity = 32;
+  attribute char_string<16> lampType = 48;
```
Comment (andy31415) [inline examples/all-clusters-app/all-clusters-common/all-clusters-app.matter:3079]: How did we decide which of the optional vs mandatory attributes we support?

All these lap settings attributes are optional and are here, however things like intrinsicBallastFactor are optional and are not here.

**Review by andy31415 — APPROVED** [review]

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.matter (hunk 2 above)
Comment (markus-becker-tridonic-com) [reply examples/all-clusters-app/all-clusters-common/all-clusters-app.matter:3079]: I changed it according to @bzbarsky-apple comment above. Everything on in all-clusters-app, only mandatory attributes on in all-clusters-minimal-app.

Attributes PowerOnLevel and PowerOnFadeTime are deprecated.

The ha.xml had this wrongly named Intrinsic**Balance**Factor. Fixed.

**Review by jtung-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@markus-becker-tridonic-com If you get a chance to merge https://github.com/markus-becker-tridonic-com/connectedhomeip/pull/1 into this PR before it merges to master, that would be great.  If not, just let me know and I will do that as a followup.

**Pull request merged**